bidict/_defaultbidict.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+# Copyright 2009-2020 Joshua Bronson. All Rights Reserved.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""Provides :class:`defaultbidict`, bidirectional defaultdict"""
+
+from collections.abc import Hashable
+from ._mut import MutableBidict
+from ._delegating import _DelegatingMixin
+
+
+class defaultbidict(_DelegatingMixin, MutableBidict):  # noqa: N801; pylint: disable=invalid-name
+    """Bidirectional defaultdict"""
+    __slots__ = ('_default_factory', )
+
+    def __init__(self, *args, **kw):
+        """Bidirectional defaultdict.
+        Provide :attr:`default` to set default factory"""
+        default_factory = kw.pop('default', None)
```

Comment:
This does not correctly handle the case where `default` is passed in as a positional arg rather than a keyword arg. (It is not a keyword-only arg.)

Reply:
The following example with `defaultdict` demonstrates how this would implementation behave incorrectly by comparison:
```python
>>> d1 = defaultdict(list, {0: [], 1: []})
>>> d1
defaultdict(<class 'list'>, {0: [], 1: []})
>>> d2 = defaultdict(list, zero=[], one=[])
>>> d2
defaultdict(<class 'list'>, {'zero': [], 'one': []})
>>> d3 = defaultdict(default_factory=list, zero=[], one=[])  # oops
>>> d3
defaultdict(None, {'default_factory': <class 'list'>, 'zero': [], 'one': []})
```